src/platform/nrfconnect/wifi/ConnectivityManagerImplWiFi.cpp

```diff
@@ -39,124 +39,200 @@ using namespace ::chip::TLV;
 namespace chip {
 namespace DeviceLayer {
 
-ConnectivityManager::WiFiStationMode ConnectivityManagerImplWiFi::_GetWiFiStationMode(void)
+CHIP_ERROR ConnectivityManagerImplWiFi::InitWiFi()
 {
-    return ConnectivityManager::WiFiStationMode::kWiFiStationMode_Disabled;
+    return WiFiManager::Instance().Init();
 }
 
-CHIP_ERROR ConnectivityManagerImplWiFi::_SetWiFiStationMode(ConnectivityManager::WiFiStationMode val)
+ConnectivityManager::WiFiStationMode ConnectivityManagerImplWiFi::_GetWiFiStationMode(void)
 {
+    if (mStationMode != ConnectivityManager::WiFiStationMode::kWiFiStationMode_ApplicationControlled)
+    {
+        mStationMode = (WiFiManager::StationStatus::DISABLED == WiFiManager().Instance().GetStationStatus())
+            ? ConnectivityManager::WiFiStationMode::kWiFiStationMode_Disabled
+            : ConnectivityManager::WiFiStationMode::kWiFiStationMode_Enabled;
+    }
+    return mStationMode;
+}
+
+CHIP_ERROR ConnectivityManagerImplWiFi::_SetWiFiStationMode(ConnectivityManager::WiFiStationMode aMode)
+{
+    VerifyOrReturnError(ConnectivityManager::WiFiStationMode::kWiFiStationMode_NotSupported != aMode,
+                        CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
+
+    if (aMode != mStationMode)
+    {
+        mStationMode = aMode;
+        if (mStationMode != ConnectivityManager::WiFiStationMode::kWiFiStationMode_ApplicationControlled)
+        {
+            bool doEnable{ ConnectivityManager::WiFiStationMode::kWiFiStationMode_Enabled == mStationMode };
+            if (doEnable)
+            {
+                OnStationConnected();
```

Comment:
Shouldn't it cause `Connect`?

Reply:
I was wondering what the `enable/disable station` actually means and how it corresponds with the connected/disconnected state. Can the station be both enabled and disconnected at the same time? Other platforms call `SetWiFiStationMode(enable)` to actually start the connect procedure by pinging the state machine inside ConnectivityManager implementation. In our case it's simpler because we directly call the WiFiManager::Connect() from WiFi driver and call the SetWiFiStationMode(enabled) only when the connection is really established. So I would rather remove the `DisconnectStation()` from line 75 to be more coherent and keep the SetWiFiStationMode for use cases when the connection/disconnection event really happens - it will make more sense when we have the proper wpa_supplicant callback API.